src/iterator.rs

```diff
@@ -303,9 +554,16 @@ impl Signals {
     ///
     /// Note that the blocking is done in this method, not in the iterator.
     pub fn wait(&self) -> Pending {
-        self.flush(true);
-
-        Pending(self.waker.pending.iter().enumerate())
+        match self.0.wait(&mut Self::has_signals) {
```

Comment:
Again, why the `&mut` in here?

Reply:
This &mut is required for the FnMut trait bound in the wait method. The FnMut is due to the adapter crate for the new [AsyncRead](https://docs.rs/futures/0.2.1/futures/io/trait.AsyncRead.html) which requires a &mut Context as one of its arguments. The mutable Context reference should be enclosed in the callback and therefore the type bound must be FnMut.